corehq/apps/domain/utils.py

```diff
@@ -124,3 +124,22 @@ def get_serializable_wire_invoice_general_credit(general_credit):
         }]
 
     return []
+
+
+@quickcache(['domain_name'])
+def get_allowed_ucr_expressions(domain_name):
+    try:
+        ucr_expressions_obj = AllowedUCRExpressionSettings.objects.get(domain=domain_name)
+        allowed_ucr_expressions = ucr_expressions_obj.allowed_ucr_expressions
+    except AllowedUCRExpressionSettings.DoesNotExist:
+        allowed_ucr_expressions = get_default_ucr_expressions()
+    return allowed_ucr_expressions
+
+
+def save_allowed_ucr_expressions(domain_name, expressions):
```

Comment:
this very much seems like it should be a `classmethod` on `AllowedUCRExpressionSettings`. also, is it worth adding a test? Are we planning to audit any of these changes?

Reply:
I have added tests for `get_allowed_ucr_expressions` https://github.com/dimagi/commcare-hq/blob/1992eed95179e6b989418415d6885a159e9e8d21/corehq/apps/domain/tests/test_utils.py#L78-L85

The`save_allowed_ucr_expressions ` seemed like a simple Model query so I don't think it requires any tests.

I was also thinking of moving them to the model class first. The reason I kept it in utils was that both of the functions were only used in forms and views, so decided to go with the utility function.

I would be happy to move it to the model class if that suits it better.

Reply:
> Are we planning to audit any of these changes?

I missed this earlier. I am not sure about it but it would be good if we do it. 
Should I add it?

Reply:
i think this might be a good question for the auditablity pod (@joxl) but I feel that might be a good idea?

Reply:
It would be good. I looked into logging the changes but the only place where I could feel it could be done was in `post`
See https://github.com/dimagi/commcare-hq/pull/31103/commits/90d07a1ce66074997aa1cef4452b485cd375844b  https://github.com/dimagi/commcare-hq/pull/31103/commits/e0789ad3f6c996cde91a5edb0763a4691006c716